README.rst

```diff
@@ -67,11 +115,10 @@ Available settings:
   - If set to ``SlotPolicy.PER_DOMAIN``, then consider setting ``SCHEDULER_PRIORITY_QUEUE = 'scrapy.pqueues.DownloaderAwarePriorityQueue'``
   to make better usage of AutoExtract concurrency and avoid delays.
 
-Within the spider, consuming the AutoExtract result is as easy as::
-
-    def parse(self, response):
-        yield response.meta['autoextract']
+Provider settings
+-----------------
 
+- ``AUTOEXTRACT_MAX_QUERY_ERROR_RETRIES`` [optional] Max number of retries for Query-level errors. Defaults to ``3``.
 
 Limitations
```

Comment:
I think it'd be nice to mention some of the limitations of the Scrapy middleware, as compared to the scrapy-poet provider:
* 429 errors are handled as standard retries when using Scrapy middleware, but they're handled properly&automatically with scrapy-poet, as it relies on scrapinghub-autoextract. One may loose some responses with the middleware. With scrapy-poet there is no need to change RETRY_HTTP_CODES.
* retrying of query-level errors is only supported in scrapy-poet integration (?)
* retries overall have a better behavior with scrapy-poet integration
* with scrapy-poet integration retry requests don't go through Scrapy
* not all data types are supported with scrapy-poet, currently only article and product are supported

Reply:
Those are good comments that are worth mentioning in our limitations section. I've just pushed a new commit adding this information. Could you check them, please? If you have further suggestions or would like to change the way something is written, please, feel free to submit a commit to this branch making these changes to avoid back and forth.